### ISSUE TITLE
On Python3, error: builtins.NameError: name 'xrange' is not defined

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8498]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8498|
|Type|enhancement|
|Created|2016-06-24 20:06:44Z|

When trying to run more of the tests under Python 3,
this error occurs in a few places:

```
builtins.NameError: name 'xrange' is not defined
```

<details><summary>Searchable metadata</summary>

```
trac-id__8498 8498
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1466798804348963 1466798804348963
changetime__1466807191858639 1466807191858639
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/214

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Looks good. Please merge. Thanks!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set status to `closed`|
|-|-|

In changeset ff5eee0c6f401e37be0ed51bb1a666314df2e6f2
```
#!CommitTicketReference repository="" revision="ff5eee0c6f401e37be0ed51bb1a666314df2e6f2"
Merge 8498-rodrigc-xrange: Use twisted.python.compat.xrange for Python 3 compatibility

Author: rodrigc
Reviewer: adiroiban
Fixes: #8498
```
